### PR TITLE
Bringing Google stats back - Reverting #3148

### PR DESCRIPTION
**Pull request by limonspb**
Google Analytics provides some nice information.
It needs some touch here and there, but it's been pretty solid.
Revert of this PR: https://github.com/betaflight/betaflight-configurator/pull/3148

Reverting had some conflicts in:
```
firmware_flashed.js
main.js
motors.js
power.js
```

i think i got them correctly

cmon, this is nice :)
image

### CODE REVIEW

**Comment by haslinghuis** [comment]
Cloud API will provide more and more data over time and users won't be able to change for example radio protocol or motor protocol as this is still possible but should not be changed.. We need to lock these settings with cloud build.

For some metrics GA could still be useful.

**Comment by limonspb** [comment]
> Cloud API will provide more and more data over time and users won't be able to change for example radio protocol or motor protocol as this is still possible but should not be changed.. We need to lock these settings with cloud build.
> 
> For some metrics GA could still be useful.

cloud API dada is cool, but it can't replace google analytics in full just yet.
Unique users per day, configurator startup per day, stats by country, fancy plots.

File: src/js/tabs/power.js
Comment (haslinghuis) [inline src/js/tabs/power.js:439]: :space_invader:

File: src/js/tabs/power.js
Comment (haslinghuis) [inline src/js/tabs/power.js:443]: :space_invader:

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by limonspb** [comment]
Added statistics collection disclaimer on the landing page as per @blckmn request.

File: src/js/Analytics.js
Comment (chmelevskij) [inline src/js/Analytics.js:13]: Why not esm class here? `prototype` based classes are relic of the past. VScode can refactor that for you.

File: src/js/main.js
Comment (chmelevskij) [inline src/js/main.js:111]: If we adding this, can this at least be moved into separate file/package. `main.js` is for bootstrapping the app and this just adds noise to what's really is important here.

File: src/js/Analytics.js
Comment (chmelevskij) [inline src/js/Analytics.js:3]: Please use modules for new code and import where appropriate.

`strict` is not needed when using modules it's a default mode.

**Comment by limonspb** [comment]
@chmelevskij this is not a new code. This is just reverting a PR.
Further improvements could be done in the separate PR

**Review by chmelevskij — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

File: src/package.json
Comment (blckmn) [inline src/package.json:1]: is this package.json superfluous?

**Review by blckmn — COMMENTED** [review]
just want to check the extra package.json in ./src?

**Comment by haslinghuis** [comment]
As GA will be deprecated in 6 months we should rather focus attention on cloud statistics and other integrations.

We also need focus on replacing deprecated chrome API's, moving to uniform UI library etc.

We already have rough ideas about which telemetry is required for analysis and configuration management integration.

File: src/package.json
Comment (limonspb) [reply src/package.json]: my bad, removed

**Comment by limonspb** [comment]
> just want to check the extra package.json in ./src?

removed. My bad :)

**Comment by limonspb** [comment]
> As GA will be deprecated in 6 months we should rather focus attention on cloud statistics and other integrations.
> 
> We also need focus on replacing deprecated chrome API's, moving to uniform UI library etc.
> 
> We already have rough ideas about which telemetry is required for analysis and configuration management integration.

There is no real focus here, just reverting PR.
After that focus can be on the new google analytics. Cloud stats is good, but i don't feel like using it for the UI user experience is a good choice, no need to invent the wheel, cuz google exists already

File: src/package.json
Comment (haslinghuis) [reply src/package.json]: This seems not right - don't rollback package,json

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Comment by limonspb** [comment]
damn found more merging issues. Need some touch.
Plz in future lets not remove the code thats working

File: src/js/Analytics.js
Comment (haslinghuis) [reply src/js/Analytics.js:3]: Need to address this

**Comment by limonspb** [comment]
> Deprecated in 6 months, rather spent effort thinking about how to do this in cloud API.

you keep missing the point about cloud API vs google stats
ANd better spend efforts on fixing something rather removing :)

**Comment by haslinghuis** [comment]
> And better spend efforts on fixing something rather than removing :)

It was discussed in Slack, you keep slowing down progression with backpedaling :rofl: 
We are just slow on development due to other issues :smiley:

**Comment by limonspb** [comment]
> > And better spend efforts on fixing something rather than removing :)
> 
> It was discussed in Slack, you keep slowing down progression with backpedaling 🤣 We are just slow on development due to other issues 😃

looks like lots of people were surprized it was removed.

**Comment by haslinghuis** [comment]
Not a lot had access so that statement is false. Common let's do this in cloud api instead as we all like nice graphs.

**Comment by limonspb** [comment]
> Not a lot had access so that statement is false. Common let's do this in cloud api instead as we all like nice graphs.

You can reinvent the wheel and track all the UI interactions, buttons, tab switching in the cloud API.
But how long it take. Instead can use something thats already developed for us for free.

**Comment by haslinghuis** [comment]
Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.

Simplistic: Options are included. Just need to read them from build and apply.

**Comment by limonspb** [comment]
> Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> 
> Simplistic: Options are included. Just need to read them from build and apply.

unique daily users, number of times configurator was launched, language selection and countries, tabs usage, etc.
All that with nice plots and a ready UI interface.

**Comment by limonspb** [comment]
> Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> 
> Simplistic: Options are included. Just need to read them from build and apply.

ok so this problem is in Master, so far i don't understand how it should be:
image

`processUID()` function gets `FC.CONFIG.uniqueDeviceIdentifier` as `undefined` in the master branch. Was that removed at some point? Maybe @blckmn knows?
Here: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/serial_backend.js#L405

**Comment by blckmn** [comment]
> > Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> > Simplistic: Options are included. Just need to read them from build and apply.
> 
> ok so this problem is in Master, so far i don't understand how it should be: image
> 
> `processUid()` function gets `FC.CONFIG.uniqueDeviceIdentifier` as `undefined` in the master branch + 4.4-zulu. Was that removed at some point? Maybe @blckmn knows? Here: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/serial_backend.js#L405

This might be related to the modules update ... or something I missed.

**Comment by limonspb** [comment]
> > > Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> > > Simplistic: Options are included. Just need to read them from build and apply.
> > 
> > 
> > ok so this problem is in Master, so far i don't understand how it should be: image
> > `processUid()` function gets `FC.CONFIG.uniqueDeviceIdentifier` as `undefined` in the master branch + 4.4-zulu. Was that removed at some point? Maybe @blckmn knows? Here: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/serial_backend.js#L405
> 
> This might be related to the modules update ... or something I missed.

ok cool, i see its configurator-related, cuz 10.8 shows correct ID.
Ill bisect later tomorrow.

**Comment by haslinghuis** [comment]
Good catch

**Comment by blckmn** [comment]
> > > > Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> > > > Simplistic: Options are included. Just need to read them from build and apply.
> > > 
> > > 
> > > ok so this problem is in Master, so far i don't understand how it should be: image
> > > `processUid()` function gets `FC.CONFIG.uniqueDeviceIdentifier` as `undefined` in the master branch + 4.4-zulu. Was that removed at some point? Maybe @blckmn knows? Here: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/serial_backend.js#L405
> > 
> > 
> > This might be related to the modules update ... or something I missed.
> 
> ok cool, i see its configurator-related, cuz 10.8 shows correct ID. I'll bisect later tomorrow.

image

it should be set if the MSP message is successful.

**Comment by limonspb** [comment]
> > > > > Much interaction won't make sense anymore with cloud build. For example radio protocol and motor protocol and other drivers are now isolated, which mean we have to lock this settings in configurator as they are predetermined by cloud build.
> > > > > Simplistic: Options are included. Just need to read them from build and apply.
> > > > 
> > > > 
> > > > ok so this problem is in Master, so far i don't understand how it should be: image
> > > > `processUid()` function gets `FC.CONFIG.uniqueDeviceIdentifier` as `undefined` in the master branch + 4.4-zulu. Was that removed at some point? Maybe @blckmn knows? Here: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/serial_backend.js#L405
> > > 
> > > 
> > > This might be related to the modules update ... or something I missed.
> > 
> > 
> > ok cool, i see its configurator-related, cuz 10.8 shows correct ID. I'll bisect later tomorrow.
> 
> image
> 
> it should be set if the MSP message is successful.

I see so it should be just devideIdentifier, not unique deviceIdentifier.

**Comment by haslinghuis** [comment]
Yes line 414 fixing now

**Comment by blckmn** [comment]
> Yes line 414 fixing now

https://github.com/betaflight/betaflight-configurator/pull/3212

**Comment by haslinghuis** [comment]
https://github.com/betaflight/betaflight-configurator/pull/3213 :joy:

**Comment by limonspb** [comment]
you guys are rockets lol

**Comment by limonspb** [comment]
ready for review/merge @haslinghuis  @chmelevskij  @blckmn

**Review by haslinghuis — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
